src/lib/isLocale.js

```diff
@@ -0,0 +1,446 @@
+import assertString from './util/assertString';
+
+/* eslint-disable max-len */
+const locales = {
+  af_NA: 'Afrikaans (Namibia)',
```

Comment:
Possible to come up with a general regex for this, since we're bound to have more locales coming up. I see for instance `en_KE` is left out.

Reply:
Alright, no problem. I can come up with it, though it might not be a guarantee that we will get them. (In case I find a unique way of identifying a locale, sure, will do the regex).
 
Alternatively, let me try find as many possible locales as possible for the same. And add them. I will try use some more resources such as [this](https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-lcid/a9eac961-e77d-41a6-90a5-ce1a8b0cdb9c)

Reply:
@profnandaa, I think after reading [this StackOverflow discussion](https://stackoverflow.com/questions/8758340/regex-to-detect-locales), 
I will go ahead to make use of regex. Thanks for your thoughts. 

```
What locale could consists of:

1.  Language code: ISO 639 2 or 3, or 4 for future use, alpha.
2. Optional script code: ISO 15924 4 alpha or 3 digit.
3. Optional country code: ISO 3166-1 2 alpha or 3 digit.
4. Separated by underscores or dashes.

Valid examples are:

- de
- en-US
- zh-Hant-TW
- En-au
- aZ_cYrl-aZ.
```

You can add label `needs update` on this one.